dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -1226,40 +1226,55 @@ public String getDefaultBranch(String repositoryId) {
     }
 
     /**
-     * Detect branches that include a .dockstore.yml using a variety of heuristics.
-     * Heuristics include the default branch, branches that are likely active or important, and branches with recent pushes.
+     * Detect references that include a .dockstore.yml.
+     */
+    public List<String> detectDockstoreYml(String repositoryId, List<String> references) {
+        try {
+            if (repositoryId == null) {
+                return List.of();
+            }
+            // Get repository.
+            GHRepository repository = getRepository(repositoryId);
+            // For a subset of references, return the corresponding refs that contain a .dockstore.yml.
+            return references.stream()
+                .filter(ref -> hasDockstoreYml(repository, ref))
+                .toList();
+
+        } catch (GHException e) {
+            LOG.error("Unable to check for .dockstore.ymls for repository " + repositoryId, e);
+            return List.of();
+        }
+    }
+
+    /**
+     * Lists the branches in the given repository in descending order of "importance".
+     * Heuristics include the default branch, branches that have names corresponding to development or main branches, and branches with recent pushes.
```

Comment:
Not clear from phone, could be code comment. Avoiding unbounded makes sense to me but what is the limit that we're trying to enforce. It looks like it used to be five, what is the the new limit?

Reply:
Limit is still five, had to refactor some code, and it made sense to move the enforcement elsewhere.